Compiler/src/ssair/passes.jl

```diff
@@ -2575,10 +2575,8 @@ function cfg_simplify!(ir::IRCode)
                     (; ssa_rename, late_fixup, used_ssas, new_new_used_ssas) = compact
                     ssa_rename[i] = SSAValue(compact.result_idx)
                     already_inserted = function (i::Int, val::OldSSAValue)
-                        if val.id in old_bb_stmts
-                            return val.id <= i
-                        end
-                        return bb_rename_pred[phi.edges[i]] < idx
+                        val.id in old_bb_stmts && return false
+                        return 0 < bb_rename_pred[phi.edges[i]] < idx
```

Comment:
The cause of the bug here was that a block was optimized away, causing `bb_rename_pred[phi.edges[i]]` to return `-3 < idx`.  I think this fix is reasonable, so we are now more conservative.

Reply:
I agree that this fix is ok, but it does cause us to look at values that we are about to drop anyway. Why not pull the loop body out of `process_phinode_values` into a separate function and call it in the loop below?

Reply:
Seems like a good idea, yes. Then in cases where we were calling `kill_current_use!(compact, val)`, we would instead not process the corresponding values and skip calling `kill_current_use!`, am I understand it correctly?

Reply:
Yes, that's correct.